tests/syscall/mount/mount-immutables-chroot.bats

```diff
@@ -0,0 +1,663 @@
+#!/usr/bin/env bats
+
+#
+# Verify trapping & emulation on "mount" and "unmount2" syscalls
+#
+
+load ../../helpers/run
+load ../../helpers/syscall
+load ../../helpers/docker
+load ../../helpers/environment
+load ../../helpers/mounts
+load ../../helpers/sysbox-health
+
+skipTest=0
+
+function teardown() {
+  sysbox_log_check
+}
+
+# Test to verify immutable mounts within a sys container's chroot() context.
+#
+# Note: a sys container immutable mount is a mount that is setup at container
+# creation time.
+#
+
+chrootpath="/tmp/chrootdir"
+immutable_ro_dir="ro_dir"
+immutable_ro_file="ro_file"
+immutable_masked_dir="masked_dir"
+immutable_masked_file="masked_file"
+
+immutable_ro_dir_path="${chrootpath}/${immutable_ro_dir}"
+immutable_ro_file_path="${chrootpath}/${immutable_ro_file}"
+immutable_masked_dir_path="${chrootpath}/${immutable_masked_dir}"
+immutable_masked_file_path="${chrootpath}/${immutable_masked_file}"
+
+function local_rootfs_prepare() {
+  mkdir -p ${immutable_ro_dir_path}
+  touch ${immutable_ro_file_path}
+  mkdir -p ${immutable_masked_dir_path}
+  touch ${immutable_masked_file_path}
+}
+
+# Testcase #1.
+#
+# Ensure immutable mounts can't be unmounted from inside the container if, and
+# only if, sysbox-fs is running with 'allow-immutable-unmounts' option disabled.
+# Alternatively, verify that unmounts are always allowed.
+@test "immutable mount can't be unmounted -- chroot()" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local_rootfs_prepare
+
+  local syscont=$(docker_run --rm -v ${immutable_ro_dir_path}:${immutable_ro_dir_path}:ro -v ${immutable_ro_file_path}:${immutable_ro_file_path}:ro --mount type=tmpfs,destination=${immutable_masked_dir_path} -v /dev/null:${immutable_masked_file_path} ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+
+  chroot_prepare ${syscont} ${chrootpath}
+
+  local immutable_mounts=$(list_container_mounts ${syscont} "0" ${chrootpath})
+  run is_list_empty ${immutable_mounts}
+  [ "$status" -ne 0 ]
+
+  # Determine the mode in which to operate.
+  local unmounts_allowed
+  run allow_immutable_unmounts
```

Comment:
Would this work: `allow_immutable_mounts=$(unmounts_allowed)` 

(It's more concise)

Or alternatively:

```
run allow_immutable_unmounts
unmounts_allowed=$(status)
```